src/lib/progress-spinner/progress-spinner.ts

```diff
@@ -84,189 +88,115 @@ export const _MdProgressSpinnerMixinBase = mixinColor(MdProgressSpinnerBase, 'pr
   encapsulation: ViewEncapsulation.None,
   preserveWhitespaces: false,
 })
-export class MdProgressSpinner extends _MdProgressSpinnerMixinBase
-    implements OnDestroy, CanColor {
+export class MdProgressSpinner extends _MdProgressSpinnerMixinBase implements CanColor,
+  OnChanges, AfterViewInit {
 
-  /** The id of the last requested animation. */
-  private _lastAnimationId: number = 0;
-
-  /** The id of the indeterminate interval. */
-  private _interdeterminateInterval: number | null;
-
-  /** The SVG <path> node that is used to draw the circle. */
-  @ViewChild('path') private _path: ElementRef;
-
-  private _mode: ProgressSpinnerMode = 'determinate';
   private _value: number;
+  private readonly _baseSize = 100;
+  private readonly _baseStrokeWidth = 10;
 
-  /** Stroke width of the progress spinner. By default uses 10px as stroke width. */
-  @Input() strokeWidth: number = PROGRESS_SPINNER_STROKE_WIDTH;
+  /** The width and height of the host element. Will grow with stroke width. **/
+  _elementSize = this._baseSize;
 
-  /**
-   * Values for aria max and min are only defined as numbers when in a determinate mode.  We do this
-   * because voiceover does not report the progress indicator as indeterminate if the aria min
-   * and/or max value are number values.
-   */
-  get _ariaValueMin() {
-    return this.mode == 'determinate' ? 0 : null;
-  }
+  /** Tracks diameters of existing instances to de-dupe generated styles. */
+  static diameters = new Set<number>([100]);
 
-  get _ariaValueMax() {
-    return this.mode == 'determinate' ? 100 : null;
+  /** The diameter of the progress spinner (will set width and height of svg). */
+  @Input()
+  get diameter(): number {
+    return this._diameter;
   }
 
-  /** @docs-private */
-  get interdeterminateInterval() {
-    return this._interdeterminateInterval;
-  }
-  /** @docs-private */
-  set interdeterminateInterval(interval: number | null) {
-    if (this._interdeterminateInterval) {
-      clearInterval(this._interdeterminateInterval);
+  set diameter(size: number) {
+    this._setDiameterClass(size);
+    this._diameter = size;
+    if (!MdProgressSpinner.diameters.has(this.diameter)) {
```

Comment:
Since we don't support the spec animation on IE and Edge, we shouldn't append the keyframes either.

Reply:
Good call